### DISCUSSION TITLE
loader not found: Texture2D

### DISCUSSION THREAD

**Discussion by VirusPC** [galacean/discussion#112] [Engine] [answered]
I have tried to load an Texture2D with resourceManager but failed. It reports error: 'loader not found: Texture2D'. 
Is I import the package in a wrong way? what can I do to resolve this problem? 

Here are the codes and error:
![image](https://github.com/galacean/discussion/assets/30697592/c11fcc7c-d6ec-4a41-b643-6a775a0a04c2)
![image](https://github.com/galacean/discussion/assets/30697592/bfa6a5b8-f1b5-4cf9-9b49-3afee7a3bd48)
![image](https://github.com/galacean/discussion/assets/30697592/f569c9cc-907a-44f2-85b0-4aa1dfa41920)

Here are the dependencies:
![image](https://github.com/galacean/discussion/assets/30697592/b78d6283-9798-4ec8-af2a-f0629da01d8c)

**Comment by gz65555** [comment, marked as answer]
You can only use `@galacean/engine` as a dependency module and remove `@galacean/engine-core`, `@galacean/engine-loader`...

**Comment by VirusPC** [comment]
Can i import sub-packages separately rather than a complete `engine`?
 I want to import them separately, but some compile error happens in `engine-physics-lite` that cannot find `@galacean/engine` module. It seems that I can only use `engine-physics-lite` with a complete `engine`?
![image](https://github.com/galacean/discussion/assets/30697592/f9ae37f1-e766-4b7b-ae42-37885dd6c376)

**Reply by gz65555** [reply]
Yes, `engine-physics-lite` and `engine-physics-physx` are separate modules. You can pick one of them as a physics backend.

**Reply by eyworldwide** [reply]
`@galancean/engine ` is the basic package(**monorepo**) you must import, which includes:
- `@galacean/engine-core`
- `@galacean/engine-loader`
- `@galacean/engine-math`
- `@galacean/engine-rhi-webgl`
- `@galacean/engine-design`

`engine-physics-lite` and `engine-physics-physx` are separate packages.
